http/transport-component.md

```diff
@@ -0,0 +1,121 @@
+# HTTP Transport Component <!-- omit in toc -->
+
+| Lifecycle Stage | Maturity      | Status | Latest Revision |
+| --------------- | ------------- | ------ | --------------- |
+| 1A              | Working Draft | Active | r0, 2023-05-31  |
+
+Authors: [@marcopolo]
+
+Interest Group: [@marcopolo], [@mxinden], [@marten-seemann]
+
+[@marcopolo]: https://github.com/marcopolo
+[@mxinden]: https://github.com/mxinden
+[@marten-seemann]: https://github.com/marten-seemann
+
+## Table of Contents <!-- omit in toc -->
+- [Context](#context)
+- [What is an HTTP transport](#what-is-an-http-transport)
+- [Multiaddr representation](#multiaddr-representation)
+- [HTTP Paths](#http-paths)
+
+
+## Context
+
+This document is only about advertising support for an HTTP transport. It
+doesn't make any assertions about how libp2p should interact with that
+transport. That will be defined in a future document.
+
+This exists to clarify the role of the `/http` component in Multiaddrs early to
+avoid confusion and conflicting interpretations.
+
+## What is an HTTP transport
+
+An HTTP transport is simply a node that can speak some standardized version of
+HTTP that is at least HTTP/1.1. Currently that means HTTP/1.1, HTTP/2, and
+HTTP/3. Intuitively if you can `curl` it with HTTP, then it speaks HTTP.
+
+Most environments will have a way to create an HTTP Client and Server, and the
+specific HTTP version used will be opaque. The client will negotiate (or [learn
+about](https://www.rfc-editor.org/rfc/rfc9114.html#section-3.1.1)) the specific HTTP version to use when communicating
+with the HTTP server.
+
+For this opaque case, we use the `/http` component at the end of the multidadr.
+If a node wants to advertise specific HTTP versions it may advertise a
+`/http-1.1`, `/http-2`, or `/h3` instead. Although these components are not
```

Comment:
Why would you do that? HTTP has its own version negotiation mechanism (ALPN for HTTP/1.1 and HTTP/2, more complicated (see above) for HTTP/3. I don't think we need to do anything here.

Reply:
You can avoid paying an extra round trip for version negotiation. If your server prefers HTTP/2 and always upgrades clients to that, you can tell clients ahead of time to skip the round trip cost for the upgrade to HTTP/2 and just start with http/2.

Reply:
There’s no round trip cost. ALPN negotiation happens during the TLS handshake.

If you made the argument that SVCB contains a protocol hint, that would make me reconsider. Although we could now end up in the awkward situation where DNS tells us one thing and the (/dns) multiaddr another.

Anyway, at this point I think we don’t really have a problem we’re solving, this seems like premature optimization.

Reply:
> There’s no round trip cost. ALPN negotiation happens during the TLS handshake.

Yes, you're correct. I was thinking of the Upgrade header which would only happen in `h2c` which would be the rare case (e.g. doing HTTP not on tls).

I'll remove this section.